cleanurl.go

```diff
@@ -0,0 +1,15 @@
+package shuttle
+
+import "net/url"
+
+// extractCredentials extracts and scrubs basic auth credentials from a URL to
+// ensure that they never get logged.
+func extractCredentials(uri *url.URL) (cleanURL *url.URL, username string, password string) {
```

Comment:
If extractCredentials took a string it would remove the need to reconstruct the String() and then re-parse it. And extractCredentials could return an error if the Parse fails.